pkg/asset/fsck.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package asset
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// detectFileSystemType uses blkid to identify the file system type of a given image file
+func detectFileSystemType(imagePath string) (string, error) {
+	cmd := exec.Command("blkid", "-o", "value", "-s", "TYPE", imagePath)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to detect file system type: %v", err)
+	}
+	return string(output), nil
+}
+
+// runFsck uses the fs-specific fsck subcommand to check if an image file is clean
+func runFsck(imagePath string, fsType string) (bool, error) {
+	fsckCommand := fmt.Sprintf("fsck.%s", fsType)
+	cmd := exec.Command(fsckCommand, "-n", imagePath)
+	cmd.Run()
+
+	exitCode := cmd.ProcessState.ExitCode()
```

Comment:
If we just want ok/notok, then it's better to look at the error returned by cmd.Run. IIRC if it failed to start, then cmd.ProcessState may be nil and crash.

What do we do if fsck crashes? Do we want to report that somehow? We could collect output and exit status, and then append to the log.

Reply:
I'm assuming that if fsck crashes, then the fs isn't clean.

Appending fsck logs somewhere might make sense. I haven't investigated this much for now. Let me get back to this with more thoughts